components/cards.css

```diff
@@ -0,0 +1,272 @@
+
+.image-card{
+    width: 20rem;
+    border-radius: 0.25rem;
+}
+.card-image-container{
+    padding: 0rem;
+    margin: 0rem;
+    width: 100%;
+    height: 12rem;
+    border-radius: inherit;
+}
+.card-img{
+    width: 100%;
+    height: 100%;
+    border-radius: inherit;
+}
+.card-button-container{
+    margin: 1rem 0rem;
+    display: flex;
+    gap: 1rem;
+}
+
+/*---------------------------------------Overlay Card------------------------------------------------*/
+
+.overlay-card{
+    margin-top: 1rem;
+    width: 15rem;
+    height: 18rem;
+    overflow: hidden;
+    border-radius: 0.25rem;
+    background-color:var(--color-light-blue);
+    text-align: center;
+    box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
+    position: relative; 
+}
+
+.overlay-card:hover{
+    box-shadow: 0 0 15px rgba(0, 0, 0, 0.25);
+}
+.overlay-card-image{
+    height: 100%;
+    width: 100%;
+    object-fit:fill;
+}
+.overlay-text{
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background: rgba(0, 0, 0, 0.6);
+    color: var(--color-white);
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    align-items: center;
+    justify-content: center;
+    gap: 1rem;
+    opacity: 0;
+    transition: opacity 0.25s;
+}
+.overlay-text:hover{
+    opacity: 1;
+}
+.overlay-text > * {
+    transform: translateY(20px);
+    transition: transform 0.25s;
+}
+.overlay-text:hover > * {
+    transform: translateY(0);
+}
+.overlay-img-description{
+    font-size: 1.25rem;
+    font-weight: 600;
+}
+
+/*---------------------------------------Product Card------------------------------------------------*/
+
+.product-card{
+    width: 13rem;
+    border-radius: 0.25rem;
+    background-color:var(--color-light-blue);
+    text-align: center;
+    box-shadow: 0 0 7px rgba(0, 0, 0, 0.25);
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    gap: 0rem;
+    position: relative; 
+}
+.product-card:hover{
+    box-shadow: 0 0 15px rgba(0, 0, 0, 0.25);
+}
+.product-image{
+    width: 100%;
+}
+.product-image img{
+    width: 100%;
+    object-fit: cover;
+}
+.product-name{
+    font-size: 1rem;
+    margin: 0rem;
+    padding: 0.25rem 0rem;
+}
+.product-price{
+    font-weight: 600;
+    margin: 0rem;
+    padding: 0.75rem 0rem;
+}
+.product-card .overlay-text{
+    opacity: 1;
+    transition: none;
+}
+
+/*---------------------------------------Horizontal Product Card------------------------------------------------*/
+
+.my-cart-product-card{
+    width: fit-content;
+    display: flex;
+    align-items: center;
+    gap: 1rem;
+    border-radius: 0.25rem;
+    box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
+}
+.my-cart-product-card:hover{
+    box-shadow: 0 0 15px rgba(0, 0, 0, 0.25);
+}
+.my-cart-product-image-div{
+    width: 12rem;
+}
+.my-cart-product-image-div img{
+    width: 100%;
+    object-fit: cover;
+}
+.my-cart-product-description-div{
```

Comment:
Just a suggestion, This classname is a bit longer try naming something like `cart-prod-desc`.